code/modules/tutorials/flock_tutorial.dm

```diff
@@ -0,0 +1,421 @@
+		src.must_deploy.UpdateOverlays(src.marker,"marker")
+
+	PerformAction(var/action, var/context)
+		if (action == FLOCK_ACTION_RIFT_SPAWN)
+			if (context == must_deploy)
+				return TRUE
+			else
+				return "<span class='alert'><b>You must deploy on the marked tile.</b></span>"
+		else if (action == FLOCK_ACTION_RIFT_COMPLETE)
+			src.finished = TRUE
+			return TRUE
+		return FALSE
+
+	TearDown()
+		must_deploy.UpdateOverlays(null,"marker")
+
+/datum/tutorialStep/flock/gatecrash
+	name = "Gatecrash"
+	instructions = "Your flockdrone is stuck in this room, use your Gatecrash ability to force the door open."
+
+	PerformAction(action, context)
+		if (action == FLOCK_ACTION_GATECRASH)
+			src.finished = TRUE
+			return TRUE
+
+/datum/tutorialStep/flock/move
+	name = "Move"
+	instructions = "Now move your flockdrone through the door by clicking and dragging it."
+
+	PerformAction(action, context)
+		if (action == FLOCK_ACTION_DRAGMOVE)
+			src.finished = TRUE
+			return TRUE
+
+/datum/tutorialStep/flock/control
+	name = "Control drone"
+	instructions = "Sometimes you may want to take direct control of a single drone, for combat or fine movement control. Click drag yourself over the flockdrone to take control of it."
+
+	PerformAction(action, context)
+		if (action == FLOCK_ACTION_DRONE_CONTROL)
+			src.finished = TRUE
+			return TRUE
+
+/datum/tutorialStep/flock/gather
+	name = "Gather resources"
+	instructions = "In order to convert the station around you, you are going to need resources. Pick up some items using your manipulator hand and place them into your disintigrator (equip hotkey) to break them down into resources."
+	var/amount = 30
+
+	PerformAction(action, context)
+		if (action == FLOCK_ACTION_GAIN_RESOURCES && context >= amount)
+			src.finished = TRUE
+			return TRUE
+
+/datum/tutorialStep/flock/convert_window
+	name = "Conversion"
+	instructions = "Convert the window in front of you to allow you to pass through it. Convert it by clicking on it with your nanite spray (middle) hand."
+
+	PerformAction(var/action, var/context)
+		if (action == FLOCK_ACTION_START_CONVERSION)
+			if (locate(/obj/window) in get_turf(context))
+				return TRUE
+			else
+				return "<span class='alert'><b>You must convert a window.</b></span>"
+		if (action == FLOCK_ACTION_TURF_CLAIM && locate(/obj/window) in context)
+			finished = TRUE
+			return TRUE
+
+/datum/tutorialStep/flock/floorrun
+	name = "Floor running"
+	instructions = "While controlling a flockdrone you can press your sprint key to disappear into the floor, becoming untargetable and passing through flock walls and windows. Use it to pass through the window you just converted."
+
+	PerformAction(action, context)
+		if (action == FLOCK_ACTION_FLOORRUN)
+			src.finished = TRUE
+			return TRUE
+
+/datum/tutorialStep/flock/release_drone
+	name = "Release control"
+	instructions = "Now use the eject button at the bottom right of your HUD to release control of this drone."
+
+	SetUp()
```

Comment:
Not really, we spawn the NPC upon completion so it makes sense to not do both at once

Reply:
Okay. If it feels fine when playing seems fine, just seemed it could be a bit much putting next step stuff down when you're being asked to do something unrelated